src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php

```diff
@@ -425,14 +426,16 @@ private function getOrderProducts(Order $order): OrderProductsForViewing
                 $product['unit_price_tax_excl'] :
                 $product['unit_price_tax_incl']
             ;
-            $totalPrice = Tools::ps_round($unitPrice, 2) * ($product['product_quantity'] - $product['customizationQuantityTotal']);
+            $totalPrice = Tools::ps_round($unitPrice, Context::getContext()->getComputingPrecision()) * ($product['product_quantity'] - $product['customizationQuantityTotal']);
```

Comment:
You should avoid using the context as much as possible, like ALWAYS if possible :P
And rounding should be done the later possible, you can check how it's done in the product PR: https://github.com/PrestaShop/PrestaShop/blob/03598a6060909e84f035393a16822c1b559b43b5/src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php#L427

Reply:
Done ! I didn't use the computingPrecision (since this is display), but the currency precision.